### ISSUE TITLE
Forbid to import symbols from a wrong place

### ISSUE THREAD

**Issue by orsinium** [wemake-services/wemake-python-styleguide#1195]
# Rule request

## Thesis

In python, you can import from a module any symbol that defined for it.

```python 
from typing import collections
```

Let's forbid indirect imports, except facades.

## Implementation

We should allow import symbols:

1. From where they are defined.
1. From `__init__.py` (facade).

Cases, when the facade isn't `__init__.py`, are rare (do you have examples?).

## Reasoning

1. Too many ways to import one thing
1. Difficult to refactor. You remove `import something` from a module that imported it for internal usage, and suddenly something else is broken.

Moved from https://github.com/wemake-services/wemake-python-styleguide/issues/1092#issuecomment-591928483

**Comment by sobolevn** [comment]
This is covered by `mypy`: see `implicit_reexport` from https://mypy.readthedocs.io/en/stable/config_file.html

We cannot handle this violation, because we cannot know what's behind `import something`
Since we do not resolve packages.
